### ISSUE TITLE
IndentationError can hide unmatched ')'

### ISSUE THREAD

**Issue by markkong** [ipython/ipython#13099]
Here is a small working example: Paste (either using a keyboard shortcut or %paste, but you can't type it in line by line because it returns on the first line on which the total number of close parentheses so far equals the number of open parentheses so far, so it won't let you type past "(spam,") the following into an ipython shell:
```
)
(spam,
  ham)
(eggs,
 beans)
```

I expect it to give me a SyntaxError showing the unmatched ')', but instead I get
```
IndentationError: unindent does not match any outer indentation level
```
(and it points to the last line).

(A slightly smaller working example is
```
)
(spam,
  ham)
(eggs,
 beans
```
)

**Comment by MrMino** [comment]
Indeed, this is IPython-specific behavior. It's undesirable, so I'll mark it as a bug. 

For people who'd like to fix it: to see what happens, you can call `%debug` right afterwards, and it should point you right at the place where the unwanted behavior stems from.

Reproducible on current `master`.
